### PR TITLE
Job role engagement

### PR DESCRIPTION
**Pull request by belwalshubham**
Targets #2005 

### Description
On clicking Job-Role-Engagement- card, a horizontal bar graph should appear, indicating each job role and the number of applications we have received.

- [x] I have performed a self-review of my own code.

### CODE REVIEW

**Review by P4NK4J — COMMENTED** [review]
@belwalshubham Based on our offline discussion, can we make thee scale difference to 100, and apply date range filters on this PR ?

**Review by P4NK4J — CHANGES_REQUESTED** [review]
@belwalshubham Overall the graph looks good. The date range filters seem not to be working fine. Please check
image

File: Modules/HR/Http/Controllers/Recruitment/ReportsController.php [hunk 1]
```diff
@@ -93,4 +94,31 @@ public function showReportCard()
             'todayCount' => $todayCount, 'verifiedApplicationsCount' => $verifiedApplicationCount,
         ]);
     }
+    public function bargraph(Request $request)
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/ReportsController.php:97]: Let's rename this function to something meaningful, say `jobWiseApplicationsGraph` ?

File: Modules/HR/Http/Controllers/Recruitment/ReportsController.php [hunk 2]
```diff
@@ -93,4 +94,31 @@ public function showReportCard()
...
+    {
+        $filters = $request->all();
+        $jobs = [];
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/ReportsController.php:100]: Do we need this ? I guess the next line where you are fetching all the jobs would be sufficient

File: Modules/HR/Routes/web.php [hunk 3]
```diff
@@ -50,6 +50,7 @@
             Route::get('campaigns', 'CampaignsController@index')->name('recruitment.campaigns');
             Route::get('Dailyapplicationcount', 'ReportsController@index')->name('recruitment.reports.index');
             Route::get('reportsCard', 'ReportsController@showReportCard')->name('recruitment.daily-applications-count');
+            Route::get('bargraph', 'ReportsController@bargraph')->name('recruitment.graph');
```
Comment (P4NK4J) [inline Modules/HR/Routes/web.php:53]: Rename here as well `name("Put actual route name, based on what its doing")

File: resources/js/app.js [hunk 4]
```diff
@@ -1408,6 +1412,80 @@ function barChart() {
 	});
 }
 
+function bargraph() {
```
Comment (P4NK4J) [inline resources/js/app.js:1415]: can we rename this function as well ?

File: resources/views/hr/recruitment/reportcard.blade.php [hunk 5]
```diff
@@ -12,8 +12,16 @@
...
+                
+            </a>
+        </div>
     </div>
-
 @endsection
 
 
+
+
```
Comment (P4NK4J) [inline resources/views/hr/recruitment/reportcard.blade.php:27]: let's remove these spaces

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Left some comments, else looks good.

File: Modules/HR/Http/Controllers/Recruitment/ReportsController.php (hunk 2 above)
Comment (belwalshubham) [reply Modules/HR/Http/Controllers/Recruitment/ReportsController.php:100]: I use filter for fetching the application from start date to end date ....... not jobs

Comment (belwalshubham) [reply Modules/HR/Http/Controllers/Recruitment/ReportsController.php:100]: I use filter for applications start date to end date not for jobs

File: Modules/HR/Http/Controllers/Recruitment/ReportsController.php [hunk 6]
```diff
@@ -116,7 +116,7 @@ public function jobWiseApplicationsGraph(Request $request)
             'application' => $applicationCount,
         ];
 
-        return view('hr.recruitment.BarGraph')->with([
+        return view('hr.recruitment.horizontalBarGraph')->with([
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/ReportsController.php:119]: @belwalshubham This `horizontalBarGraph` can be for many modules, let's name it specific `jobWiseApplicationsGraph` instead

File: Modules/HR/Routes/web.php [hunk 7]
```diff
@@ -50,7 +50,7 @@
             Route::get('campaigns', 'CampaignsController@index')->name('recruitment.campaigns');
             Route::get('Dailyapplicationcount', 'ReportsController@index')->name('recruitment.reports.index');
             Route::get('reportsCard', 'ReportsController@showReportCard')->name('recruitment.daily-applications-count');
-            Route::get('applications/barChart', 'ReportsController@jobWiseApplicationsGraph')->name('applications.graph');
+            Route::get('applications/barChart', 'ReportsController@jobWiseApplicationsGraph')->name('applications.job-Wise-Applications-Graph');
```
Comment (P4NK4J) [inline Modules/HR/Routes/web.php:53]:
```suggestion
            Route::get('applications/jobWiseApplicatonReport', 'ReportsController@jobWiseApplicationsGraph')->name('applications.job-Wise-Applications-Graph');
```

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Left some comments, please check

**Review by P4NK4J — APPROVED** [review]
Looks good.

**Pull request merged**